examples/hydro/hydro_cylinders.py

```diff
@@ -71,23 +71,18 @@ def main():
         lagrangian_spoke["opt_kwargs"]["all_nodenames"] = all_nodenames
 
     # xhat looper bound spoke
-    xhat_scenario_dict = {"ROOT": "Scen1",
-                          "ROOT_0": "Scen1",
-                          "ROOT_1": "Scen4",
-                          "ROOT_2": "Scen7"}
     
-    if with_xhatspecific:
-        xhatspecific_spoke = vanilla.xhatspecific_spoke(*beans,
-                                                        xhat_scenario_dict,
+    if with_xhatshuffle:
+        xhatshuffle_spoke = vanilla.xhatshuffle_spoke(*beans,
                                                         all_nodenames,
                                                         BFs,
                                                         scenario_creator_kwargs=scenario_creator_kwargs)
 
```

Comment:
```suggestion
xhatsuffle_spoke["opt_kwargs"]["all_nodenames"] = all_nodenames

```
I'm a bit surprised this works without this. For a separate PR, we should figure out the `beans` issue with `all_nodenames`

Reply:
This is related to the way vanilla treats branching factors and all_nodenames. I made a comment about this in vanilla